src/nvim/api/private/helpers.c

```diff
@@ -808,6 +811,17 @@ static void init_type_metadata(Dictionary *metadata)
 
   PUT(*metadata, "types", DICTIONARY_OBJ(types));
 }
+static void init_api_level_metadata(Dictionary *metadata)
+{
+  Dictionary version = ARRAY_DICT_INIT;
+  PUT(version, "current", INTEGER_OBJ(NVIM_API_CURRENT));
+  PUT(version, "compatibility", INTEGER_OBJ(NVIM_API_COMPATIBILITY));
+#ifdef NVIM_VERSION_PRERELEASE
+  PUT(version, "prerelease", BOOLEAN_OBJ(true));
```

Comment:
> I don't think providing extra metadata to help delineate whether the API is released or yet is necessary. It's a technical measure for a social problem.

Sigh, no one has ever claimed that this is the best solution to some specific "social" problem.  The point is simply to provide the plugin/host with _information_ so they can make a decision the way they prefer (also including, not using global information at all and instead do a deep inspection), without setting a policy for _all_ plugins.

Again, we we don't include this, we need to make a  decision to either specify the old stable level or the in-dev level,  _either_ way a delineation is being done. I've heard good arguments for both. IMO that adds up to letting api clients get all the information. But _either_ way a decision to take some "technical measure" is being made, I'm just proposing the one that forces the least assumptions on clients.